app/views/shared/_header.html.erb

```diff
@@ -0,0 +1,10 @@
+
+<% if logged_in? %>
+    <h1>What's Up, <%= link_to "#{current_user.name}", "/users/#{current_user.id}" %></h1>
+    <%= link_to "Logout", '/logout', method: :destroy%>
+<%else%>
+    <%= button_to "Login", '/login', method: :get%>
+    <%= button_to "Sign Up", '/users/new', method: :get%>
+<%end%>
```

Comment:
* [ ] There should be absolutely NO logic in views.

  > Views are only for displaying information that became available to it from its corresponding controller not generating it. **Resources**
* In [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) explains how to put if/else statements within helpers so let's do the same for your code. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright